snow/validators/retriever_test.go

```diff
@@ -0,0 +1,69 @@
+package validators
```

Comment:
Please use external tests by writing test code in a test package (`validators_test`) instead of writing internal tests. See [knowledge base](https://optakt.github.io/code/testing/#internal-unit-tests) on this.

By default, we should use external tests, unless there is a case of the following:

* We need to access the attributes of the type we are testing (in case checking outputs is not enough)
* We need to test helper functions that are not exposed publicly in the package

Here it is not the case so we should be using an external test package.